sumologic_collectd_metrics/metrics_converter.py

```diff
@@ -144,3 +145,60 @@ def convert_to_metrics(data, data_set, sep):
         metrics.append(metric)
 
     return metrics
+
+
+def process_signalfx_statsd_tags(data):
+    """
+    Process statsd tags embedded in the metric name within the supplied raw data. This function
+    modifies the raw data in-place.
+
+    :param data:
+    :type data: collectd.Values
+    :return:
+    :rtype: None
+    """
+    cleaned_type_instance, new_metadata = parse_statsd_signalfx_metric_name(
+        data.type_instance
+    )
+    data.type_instance = cleaned_type_instance
+    data.meta.update(new_metadata)
+
+
+def parse_statsd_signalfx_metric_name(metric_name):
+    """
+    Parse the metric name and extract metadata from it according to SignalFx' custom extension to
+    the StatsD protocol.
+    See: https://docs.signalfx.com/en/latest/integrations/agent/monitors/collectd-statsd.html#adding-dimensions-to-statsd-metrics
+
+    :param metric_name: The metric name
+    :type metric_name: str
+    :return: A pair consisting of the modified metric name without the metadata, and a dict of
+    extracted metadata.
+    :rtype: (str, dict)
+    """  # pylint: disable=line-too-long
+    matcher = re.compile(r"\[([^\]]+)\]")
+    matches = matcher.findall(metric_name)
+    if len(matches) != 1:
+        # if we find more than one metadata segment, something is probably wrong, and we should
+        # ignore it
+        # if we find no segments, we have nothing to do
+        return metric_name, {}
+
+    metadata = {}
+    tag_string = matches[0]
+    key_value_strings = tag_string.split(",")
+    for key_value_string in key_value_strings:
+        parts = key_value_string.split("=")
+        if len(parts) != 2:  # ignore malformed entries, but parse as much as we can
+            continue
+        key, value = [p.strip() for p in parts]
+        if key == "":
+            continue
```

Comment:
Impact if we enable support for dummy keys ?

Reply:
Can you elaborate on what you mean by "dummy keys"? I think Sumo itself will drop an empty key even if we send it here.

Reply:
was thinking if we should consider `statsd.[=bar,dim=val,=24,perf=97]test` and transform it into `statsd.test` with dictionary keys as `bar=bar` `dim=val` `24=24`

But i feel this would mess up the dataset.

Reply:
For now, can we just add a debug log here ?

Reply:
Added a warning.